pkgs/build-support/rust/replace-workspace-values.py

```diff
@@ -0,0 +1,43 @@
+import sys
+import toml
+
+crate_manifest = toml.load(sys.argv[1])
+workspace_manifest = toml.load(sys.argv[2])
+
+
+def replace_key(table, section, key):
+    if "workspace" in table[key] and table[key]["workspace"] is True:
+        print("replacing " + key)
+
+        replaced = table[key]
+        del replaced["workspace"]
+
+        workspace_copy = workspace_manifest["workspace"][section][key]
+
+        if section == "dependencies":
+            if isinstance(workspace_copy, str):
+                replaced["version"] = workspace_copy
+            else:
+                replaced.update(workspace_copy)
```

Comment:
> [`features`](https://doc.rust-lang.org/cargo/reference/features.html) declared in this table are additive with the `features` from `[dependencies]`

We should probably handle this

https://doc.rust-lang.org/cargo/reference/workspaces.html#the-dependencies-table

Reply:
Agreed, done.